File: FoodOrganizer/app/src/main/java/es/fron99/foodorganize/Adapters/AdapterListTimeMenu.java
package es.fron99.foodorganize.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import es.fron99.foodorganize.Models.Food;
import es.fron99.foodorganize.Models.Menu;
import es.fron99.foodorganize.R;

public class AdapterListTimeMenu extends RecyclerView.Adapter<AdapterListTimeMenu.ViewHolder> {

private ArrayList<Menu> menus;

public static class ViewHolder extends RecyclerView.ViewHolder {
    private final TextView txtViewNameMenu, txtViewSmallDescriptionMenu, txtViewTimePrepareMenu;

    public ViewHolder(View view) {
        super(view);
        txtViewNameMenu = view.findViewById(R.id.txtViewNameMenu);
        txtViewSmallDescriptionMenu = view.findViewById(R.id.txtViewSmallDescriptionMenu);
        txtViewTimePrepareMenu = view.findViewById(R.id.txtViewTimePrepareMenu);
    }

    public TextView getTxtViewNameMenu() {
        return txtViewNameMenu;
    }

    public TextView getTxtViewSmallDescriptionMenu() {
        return txtViewSmallDescriptionMenu;
    }

    public void setTxtViewNameMenu(String nameMenu) {
        txtViewNameMenu.setText(nameMenu);
    }

    public void setTxtViewSmallDescriptionMenu(String smallDescriptionMenu) {
        txtViewSmallDescriptionMenu.setText(smallDescriptionMenu);
    }

    public TextView getTxtViewTimePrepareMenu() {
        return txtViewTimePrepareMenu;
    }

    public void setTxtViewTimePrepareMenu(String timePrepareMenu) {
        txtViewTimePrepareMenu.setText(timePrepareMenu);
    }

}

    public AdapterListTimeMenu(ArrayList<Menu> dataSet) {
        menus = new ArrayList<>(dataSet);
    }

    @NonNull
    @Override
    public AdapterListTimeMenu.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.menu_card, viewGroup, false);
        return new AdapterListTimeMenu.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(AdapterListTimeMenu.ViewHolder viewHolder, final int position) {
        Menu menu = this.menus.get(position);
        viewHolder.setTxtViewNameMenu(menu.getName());

        String descripcion = menu.getSmallDescription()+"\n\n";

        for (Food food :menu.getFoods()) {
            descripcion += "- "+food.getName()+"\n";
        }

        viewHolder.setTxtViewSmallDescriptionMenu(descripcion);
        viewHolder.setTxtViewTimePrepareMenu(menu.getTimeToPrepare()+"MIN");
    }


    @Override
    public int getItemCount() {
        return menus.size();
    }

}

File: FoodOrganizer/app/src/main/java/es/fron99/Foodorganize/Adapters/AdapterListTimeMenu.kt
package es.fron99.Foodorganize.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import es.fron99.Foodorganize.Dao.Model.MenuWithFoods
import es.fron99.Foodorganize.R
import java.util.*


class AdapterListTimeMenu(dataSet: ArrayList<MenuWithFoods>?) : RecyclerView.Adapter<AdapterListTimeMenu.ViewHolder>() {

    private var menus: ArrayList<MenuWithFoods> = ArrayList(dataSet)

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val txtViewNameMenu: TextView = view.findViewById(R.id.txtViewNameMenu)
        val txtViewSmallDescriptionMenu: TextView = view.findViewById(R.id.txtViewSmallDescriptionMenu)
        val txtViewTimePrepareMenu: TextView = view.findViewById(R.id.txtViewTimePrepareMenu)

        fun setTxtViewNameMenu(nameMenu: String?) {
            txtViewNameMenu.text = nameMenu
        }

        fun setTxtViewSmallDescriptionMenu(smallDescriptionMenu: String?) {
            txtViewSmallDescriptionMenu.text = smallDescriptionMenu
        }

        fun setTxtViewTimePrepareMenu(timePrepareMenu: String?) {
            txtViewTimePrepareMenu.text = timePrepareMenu
        }

    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.menu_card, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val menu = menus[position]
        viewHolder.setTxtViewNameMenu(menu.menu.name)
        var descripcion = """
            ${menu.menu.smallDescription}
            
            
            """.trimIndent()

        for (food in menu.foods) {
            descripcion += """
                - ${food.name}
                
                """.trimIndent()
        }
        viewHolder.setTxtViewSmallDescriptionMenu(descripcion)
        viewHolder.setTxtViewTimePrepareMenu(calculateTime(menu.timeToPrepare))
    }

    override fun getItemCount(): Int {
        return menus.size
    }

    fun changeData(dataSet: ArrayList<MenuWithFoods>?){
        this.menus.clear()
        this.menus = ArrayList(dataSet)
        this.notifyDataSetChanged()
    }

    fun calculateTime(time: Int) : String{

        return (
                if (time <= 59){"$time MIN" }
                else{calculateHours(time)}
                )

    }

    fun calculateHours(time: Int): String{

        var timeS : String = ""
        val hours :Int = time/60
        val minuts : Int = (((time/60.toDouble()) - hours) * 60).toInt()
        timeS = "$hours H $minuts MIN"
        return timeS

    }

}